File: dotnet/CudaRS/Ocr/OpenVinoOcrModelConfig.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace CudaRS.Ocr;

public sealed class OpenVinoOcrModelConfig
{
    [JsonPropertyName("det_model_path")]
    public string DetModelPath { get; set; } = string.Empty;

    [JsonPropertyName("rec_model_path")]
    public string RecModelPath { get; set; } = string.Empty;

    [JsonPropertyName("dict_path")]
    public string DictPath { get; set; } = string.Empty;

    [JsonPropertyName("device")]
    public string? Device { get; set; } = "cpu";

    [JsonPropertyName("det_resize_long")]
    public int? DetResizeLong { get; set; } = 960;

    [JsonPropertyName("det_stride")]
    public int? DetStride { get; set; } = 128;

    [JsonPropertyName("det_thresh")]
    public float? DetThresh { get; set; } = 0.3f;

    [JsonPropertyName("det_box_thresh")]
    public float? DetBoxThresh { get; set; } = 0.6f;

    [JsonPropertyName("det_unclip_ratio")]
    public float? DetUnclipRatio { get; set; } = 1.5f;

    [JsonPropertyName("det_max_candidates")]
    public int? DetMaxCandidates { get; set; } = 1000;

    [JsonPropertyName("det_min_area")]
    public int? DetMinArea { get; set; } = 10;

    [JsonPropertyName("det_box_padding")]
    public int? DetBoxPadding { get; set; } = 2;

    [JsonPropertyName("rec_input_h")]
    public int? RecInputH { get; set; } = 48;

    [JsonPropertyName("rec_input_w")]
    public int? RecInputW { get; set; } = 320;

    [JsonPropertyName("rec_batch_size")]
    public int? RecBatchSize { get; set; } = 8;

    [JsonPropertyName("rec_score_thresh")]
    public float? RecScoreThresh { get; set; } = 0.0f;

    public string ToJson()
        => JsonSerializer.Serialize(this, new JsonSerializerOptions { DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull });
}

File: cudars-ffi/src/sdk/openvino_ocr_model_config.rs
use serde::Deserialize;

#[derive(Debug, Clone, Deserialize)]
#[serde(default)]
pub struct OpenVinoOcrModelConfig {
    pub det_model_path: String,
    pub rec_model_path: String,
    pub dict_path: String,
    pub device: Option<String>,
    pub det_resize_long: i32,
    pub det_stride: i32,
    pub det_thresh: f32,
    pub det_box_thresh: f32,
    pub det_unclip_ratio: f32,
    pub det_max_candidates: i32,
    pub det_min_area: i32,
    pub det_box_padding: i32,
    pub rec_input_h: i32,
    pub rec_input_w: i32,
    pub rec_batch_size: i32,
    pub rec_score_thresh: f32,
}

impl Default for OpenVinoOcrModelConfig {
    fn default() -> Self {
        Self {
            det_model_path: String::new(),
            rec_model_path: String::new(),
            dict_path: String::new(),
            device: Some("cpu".to_string()),
            det_resize_long: 960,
            det_stride: 128,
            det_thresh: 0.3,
            det_box_thresh: 0.6,
            det_unclip_ratio: 1.5,
            det_max_candidates: 1000,
            det_min_area: 10,
            det_box_padding: 2,
            rec_input_h: 48,
            rec_input_w: 320,
            rec_batch_size: 8,
            rec_score_thresh: 0.0,
        }
    }
}
